hal/src/nRF52840/ble_hal.cpp

```diff
@@ -3709,7 +3709,6 @@ int hal_ble_gap_connect_cancel(const hal_ble_addr_t* address, void* reserved) {
 }
 
 int hal_ble_gap_disconnect(hal_ble_conn_handle_t conn_handle, void* reserved) {
-    BleLock lk;
```

Comment:
Wouldn't this introduces thread-safety issues when accessing/traversing `connections_` ? I also saw that we specifically release wiring lock in `spark_wiring_ble.cpp` before calling `hal_ble_gap_disconnect()`, so any deadlocking issues should already be mitigated?

Reply:
You're right. We need to guard the `connections` in BLE HAL. I reverted this change and added a FIXME in `onBleLinkEvents()` in the wiring to make us be aware of a potential deadlock.